base/applications/notepad/settings.c

```diff
@@ -228,8 +236,18 @@ void NOTEPAD_SaveSettingsToRegistry(void)
 {
     HKEY hKey;
     DWORD dwDisposition;
+    WINDOWPLACEMENT wp;
+    UINT x, y, cx, cy;
+
+    wp.length = sizeof(wp);
+    GetWindowPlacement(Globals.hMainWnd, &wp);
+    x = wp.rcNormalPosition.left;
+    y = wp.rcNormalPosition.top;
+    cx = wp.rcNormalPosition.right - x;
+    cy = wp.rcNormalPosition.bottom - y;
+    if (wp.flags & WPF_RESTORETOMAXIMIZED)
+        cy |= 0x80000000;
```

Comment:
This looks convoluted. There are other applications that save their windowpos state and that can restore it when they restart, and such code isn't used at all in them. See e.g. eventvwr, amongst others.

Reply:
@katahiromz preferred using the existing registry values, otherwise I would just store the WINDOWPLACEMENT as a REG_BINARY instead.

Reply:
In principle it could work even if you store the rcNormalPosition rect + the showWnd value. As I previously mentioned eventvwr just works and it does that.
Here's for example what it retrieves from the registry: https://git.reactos.org/?p=reactos.git;a=blob;f=base/applications/mscutils/eventvwr/eventvwr.c;h=f707d58504110b9b16025498bcea8bb63da4b6ca;hb=HEAD#l480
(Granted here, the format used is via a string, to be windows-compatible, but a similar thing with notepad should be doable, using whatever format is used also by windows' own notepad.)

Reply:
> As I previously mentioned eventvwr just works and it does that.

And it is broken so telling me to use that is not helpful.

How is it broken you ask? Place the taskbar on top of the screen and open/close eventvwr a couple of times and watch it jump higher and higher on the screen each time you open it.

Allow me to quote from MSDN:

 > The coordinates used in a WINDOWPLACEMENT structure should be used only by the GetWindowPlacement and SetWindowPlacement functions. Passing workspace coordinates to functions which expect screen coordinates (such as SetWindowPos) will result in the window appearing in the wrong location. For example, if the taskbar is at the top of the screen, saving window coordinates using GetWindowPlacement and restoring them using SetWindowPos causes the window to appear to "creep" up the screen.

Reply:
Thanks, this is a point I didn't know.